rslib/src/search/parser.rs

```diff
@@ -57,6 +57,7 @@ pub enum SearchNode {
         field: String,
         text: String,
         is_re: bool,
+        is_nc: bool,
```

Comment:
Our data structure makes it appear like is_re=1 && is_nc=1 is possible, but we don't support that when parsing/writing. An enum field of normal/re/nc would better convey what we actually support.